src/Internal/Payment/Payment.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Class Payment
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment;
+
+use WC_Order;
+use WCPay\Internal\Payment\PaymentMethod\PaymentMethodInterface;
+
+/**
+ * Payment object.
+ *
+ * This is a data object, used as context and storage for payment states.
+ */
+class Payment {
+	/**
+	 * Order requiring/having payment.
+	 *
+	 * @var WC_Order
+	 */
+	private $order;
+
+	/**
+	 * Holds all data, related to the payment.
+	 *
+	 * @var array
+	 */
+	private $data = [];
+
+	/**
+	 * Payment constructor.
+	 *
+	 * @param WC_Order $order Order requiring payment.
+	 */
+	public function __construct( WC_Order $order ) {
+		$this->order = $order;
+	}
```

Comment:
> Is `Payment` like a `payment_context` we discussed? 

Yes. It is the context.

> If so, it might be worth to store scalar data only.
> Especially here, PHP will pass objects by reference, but it's a bit of a hidden feature of the language, and the order object can't be serialized

No. We will need to store various objects within the context, incl. payment intents (or whatever abstracts/wraps them) to make them accessible between states. Storing an ID, and loading whatever it is (intent, order, payment, etc.) whenever needed, is not optimal.

It's worth considering that what we have here is just working data. That's why the `Storage` class has methods to import/export the necessary data from the payment, making them scalar.

With certain objects, especially orders, we should likely strive to be consistent, and intentionally try to use the same instance. Modifying different objects, and saving them individually, might be dangerous.